cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -126,6 +132,87 @@ public CloseableIterator<StructuredRow> scan(Range keyRange, int limit) throws I
                              limit);
   }
 
+
+  /*
+   *  Sorting in memory for the no-sql implementation. We sort post table scan.
+   * */
+  @Override
+  public CloseableIterator<StructuredRow> scan(Range keyRange, int limit, String orderByField, SortOrder sortOrder)
+    throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: Scan range {} with limit {}, sort field {} and sort order {}", schema.getTableId(),
+              keyRange, limit, orderByField, sortOrder);
+    if (!schema.isIndexColumn(orderByField)) {
+      throw new InvalidFieldException(schema.getTableId(), orderByField, "is not an indexed column");
+    }
+    Comparator<StructuredRow> comparator = sortOrder.equals(SortOrder.ASC) ? getComparator(orderByField) :
+      getComparator(orderByField).reversed();
+    PriorityQueue<StructuredRow> rows = new PriorityQueue<>(comparator);
+
+    try (ScannerIterator scannerIterator = new ScannerIterator(getScanner(keyRange), schema)) {
+      while (scannerIterator.hasNext()) {
+        rows.offer(scannerIterator.next());
+      }
+    }
+    // return an iterator for the sorted elements
+    AbstractCloseableIterator<StructuredRow> abstractCloseableIterator = new
+      AbstractCloseableIterator<StructuredRow>() {
+      @Override
+      protected StructuredRow computeNext() {
+        while (!rows.isEmpty()) {
+          return rows.poll();
+        }
+        return endOfData();
+      }
+
+      @Override
+      public void close() {
+        // no-op
+      }
+    };
+    return new LimitIterator(Collections.singleton(abstractCloseableIterator).iterator(), limit);
+  }
+
+  private Comparator<StructuredRow> getComparator(String orderByField)
+    throws InvalidFieldException {
+    switch (schema.getType(orderByField)) {
+      case INTEGER:
+        return (row1, row2) ->Objects.compare(row1.getInteger(orderByField), row2.getInteger(orderByField),
```

Comment:
indentation
You could use `option+command+L` to auto format

Reply:
this indentation seems right? L176 is extra that can be moved

Reply:
yeah might be minor config thing in my IDE. It says there should be space here before Objects: `(row1, row2) -> Objects.compare(`

Reply:
Yes thats how my IDE also shows it, but I think @albertshau mentioned in a previous comment that we don't want that space

Reply:
It had two spaces before. The comment was it should have one.